precompile/contracts/nativeminter/config.go

```diff
@@ -95,3 +99,90 @@ func (c *Config) Verify(chainConfig precompileconfig.ChainConfig) error {
 	}
 	return c.AllowListConfig.Verify(chainConfig, c.Upgrade)
 }
+
+func (c *Config) MarshalBinary() ([]byte, error) {
+	keys := make([]common.Address, 0)
+	for key := range c.InitialMint {
+		keys = append(keys, key)
+	}
+
+	sort.Slice(keys, func(i, j int) bool {
+		return bytes.Compare(keys[i][:], keys[j][:]) < 0
+	})
+
+	p := wrappers.Packer{
+		Bytes:   []byte{},
+		MaxSize: 1 * units.MiB,
+	}
+
+	bytes, err := c.AllowListConfig.MarshalBinary()
+	if err != nil {
+		return nil, err
+	}
+
+	p.PackBytes(bytes)
+	if p.Err != nil {
+		return nil, p.Err
+	}
+
+	bytes, err = c.Upgrade.MarshalBinary()
+	if err != nil {
+		return nil, err
+	}
+
+	p.PackBytes(bytes)
+
+	p.PackInt(uint32(len(keys)))
+	if p.Err != nil {
+		return nil, p.Err
+	}
+
+	for _, key := range keys {
+		p.PackBytes(key[:])
+		if p.Err != nil {
+			return nil, p.Err
+		}
+		p.PackBytes((*big.Int)(c.InitialMint[key]).Bytes())
+		if p.Err != nil {
+			return nil, p.Err
+		}
+	}
+
+	return p.Bytes, nil
+}
+
+func (c *Config) UnmarshalBinary(bytes []byte) error {
+	p := wrappers.Packer{
+		Bytes: bytes,
+	}
+	allowList := p.UnpackBytes()
+	if p.Err != nil {
+		return p.Err
+	}
+	upgrade := p.UnpackBytes()
+	if p.Err != nil {
+		return p.Err
+	}
+	if err := c.AllowListConfig.UnmarshalBinary(allowList); err != nil {
+		return err
+	}
+	if err := c.Upgrade.UnmarshalBinary(upgrade); err != nil {
+		return err
+	}
+	len := p.UnpackInt()
+	c.InitialMint = make(map[common.Address]*math.HexOrDecimal256, len)
+
+	for i := uint32(0); i < len; i++ {
+		key := common.BytesToAddress(p.UnpackBytes())
```

Comment:
Can we make sure to pack addresses as 20 fixed bytes and big int values as 32 fixed bytes?